controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -408,6 +423,215 @@ func (r *ROSAControlPlaneReconciler) updateOCMCluster(rosaScope *scope.ROSAContr
+			Component(cmv1.NewClientComponent().Name(client.ComponentName).Namespace(client.ComponentNamespace)))
+	}
+	externalAuthBuilder.Clients(clientsBuilders...)
+
+	// claims builder
+	if authProvider.ClaimMappings != nil {
+		clainMappingsBuilder := cmv1.NewTokenClaimMappings()
+		if authProvider.ClaimMappings.Groups != nil {
+			clainMappingsBuilder.Groups(cmv1.NewGroupsClaim().Claim(authProvider.ClaimMappings.Groups.Claim).
+				Prefix(authProvider.ClaimMappings.Groups.Prefix))
+		}
+
+		if authProvider.ClaimMappings.Username != nil {
+			usernameClaimBuilder := cmv1.NewUsernameClaim().Claim(authProvider.ClaimMappings.Username.Claim).
+				PrefixPolicy(string(authProvider.ClaimMappings.Username.PrefixPolicy))
+			if authProvider.ClaimMappings.Username.Prefix != nil {
+				usernameClaimBuilder.Prefix(*authProvider.ClaimMappings.Username.Prefix)
+			}
+
+			clainMappingsBuilder.UserName(usernameClaimBuilder)
+		}
+
+		claimBuilder := cmv1.NewExternalAuthClaim().Mappings(clainMappingsBuilder)
+
+		validationRulesbuilders := make([]*cmv1.TokenClaimValidationRuleBuilder, 0, len(authProvider.ClaimValidationRules))
+		for _, rule := range authProvider.ClaimValidationRules {
+			validationRulesbuilders = append(validationRulesbuilders, cmv1.NewTokenClaimValidationRule().
+				Claim(rule.RequiredClaim.Claim).RequiredValue(rule.RequiredClaim.RequiredValue))
+		}
+		claimBuilder.ValidationRules(validationRulesbuilders...)
+
+		externalAuthBuilder.Claim(claimBuilder)
+	}
+
+	externalAuthConfig, err := externalAuthBuilder.Build()
+	if err != nil {
+		return fmt.Errorf("failed to build external auth config: %v", err)
+	}
+
+	if shouldUpdate {
+		_, err = externalAuthClient.UpdateExternalAuth(cluster.ID(), externalAuthConfig)
+		if err != nil {
+			return fmt.Errorf("failed to update external authentication provider '%s' for cluster '%s': %v",
+				externalAuthConfig.ID(), rosaScope.InfraClusterName(), err)
+		}
+	} else {
+		_, err = externalAuthClient.CreateExternalAuth(cluster.ID(), externalAuthConfig)
+		if err != nil {
+			return fmt.Errorf("failed to create external authentication provider '%s' for cluster '%s': %v",
+				externalAuthConfig.ID(), rosaScope.InfraClusterName(), err)
+		}
+	}
+
+	lastAppliedAnnotation, err := json.Marshal(authProvider)
+	if err != nil {
+		return err
+	}
+
+	if rosaScope.ControlPlane.Annotations == nil {
+		rosaScope.ControlPlane.Annotations = make(map[string]string)
+	}
+	rosaScope.ControlPlane.Annotations[ExternalAuthProviderLastAppliedAnnotation] = string(lastAppliedAnnotation)
+
+	return nil
+}
+
+func (r *ROSAControlPlaneReconciler) reconcileExternalAuthBootstrapKubeconfig(ctx context.Context, externalAuthClient *rosa.ExternalAuthClient, rosaScope *scope.ROSAControlPlaneScope, cluster *cmv1.Cluster) error {
+	kubeconfigSecret := rosaScope.ExternalAuthBootstrapKubeconfigSecret()
+	err := r.Client.Get(ctx, client.ObjectKeyFromObject(kubeconfigSecret), kubeconfigSecret)
+	if err == nil {
+		// already exist.
```

Comment:
Break-glass credentials have a (short) finite TTL - how do we rotate the value of this?

Reply:
the decision is to not rotate this. This will only be created once initially for the user to setup their RBAC. If needed, they could potentially just delete the secret to get a new valid credential created.

Reply:
Will we remove the secret once the content is invalid?

Reply:
no, we rely on its (non)existence to decide if we need to create a new one or not.